gemfile.d/test.rb

```diff
@@ -0,0 +1,7 @@
+group :test do
```

Comment:
Why move these into this group in a Gemfile rather than keep them where they are?

Reply:
Today, they are defined as development dependencies, but we do not install the "development" group in CI as that would be silly (we don't need pry and friends from https://github.com/theforeman/foreman/blob/develop/bundler.d/development.rb).
By moving them to a bundler.d file, we can define them as a group (you can't do that in the gemspec) and install cleanly.

Reply:
More in my old comment at https://github.com/Katello/katello/pull/10780#pullrequestreview-1752348091

Reply:
I agree and disagree. These testing frameworks are, I would argue, part of development. We shouldn't divorce testing from development.

Reply:
But are we divorcing?
`bundle install` still installs them by default, we just can avoid *pure* dev deps (pry) by using "without development" config if we want to (in CI).
And this follows the same pattern we have in core.

Reply:
Technically development dependencies in a gemspe end up in a bundler group called development. This just further split it.

It should be noted you can no longer load them via gemspec in a bundler file. Perhaps that's something that will show up in gems that depend on katello and want to use its tests

Reply:
I've opened https://github.com/theforeman/jenkins-jobs/pull/410 to at least fix Jenkins support for this setup.

Reply:
@ehelms do the recent change to move all devel-like deps to `gemfile.d/test.rb` satisfy your need for "consistency"?

Reply:
I sense snark in the force. And yes it does. Thank you.